File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<pair<int,int>> vpii;
        for(int i= 0; i<nums.size();i++){
            vpii.push_back({nums[i], i});
        }
        sort(vpii.begin(), vpii.end());
        int start = 0, end = vpii.size()-1;
        while(start<end){
            if(vpii[start].first+vpii[end].first == target){
                return {vpii[start].second, vpii[end].second};
            }
            else if(vpii[start].first+vpii[end].first > target){
                end--;
            }
            else{
                start++;
            }
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    //Brute force: check every pair
    // Time On2 Space O1
    
    // for i:= 0; i<len(nums);i++{
    //     for j := i+1; j<len(nums);j++{
    //         if nums[i]+nums[j] == target {
    //             return []int{i, j}
    //         }
    //     }
    // }
    // return nil

    seen := make(map[int]int)
    for i, n:= range nums{
        if j, ok := seen[target-n]; ok {
            return []int{i, j}
        }
        seen[n] = i
    }
    return nil

}